### PR TITLE
fix(fe-fpm-writer): Writing into manifest.json always provides 2 spaces for tab size.

### LINKED ISSUE #856: BUG - fe-fpm-writer. Writing into `manifest.json` always provides 2 spaces for tab size.
### Description
`fpm-writer` methods like `generateCustomColumn`, `generateCustomSection`, etc. updates and reformats `manifest.json` with 2 spaces for tab size. 

### Steps to Reproduce
1. Use app and format `manifest.json` in way that 4 spaces are used for tab size;
2. Trigger creation of custom column or other extension like using method `generateCustomColumn`

Problem - `manifest.json` is reformatted and uses 2 spaces for tab sizing.

image

### Expected results
Provide option to set tab size.

### Actual results
Problem - `manifest.json` is reformatted and in result uses 2 spaces for tab size.

### Screenshots
If applicable, add screenshots to help explain the problem.

#### Version/Components/Environment	
Add any other context about the problem here
OS:                      
- [x] Mac OS
- [x] Windows
- [ ] Other

### Problem
{describe the problem} 

### Fix
{describe the fix} 

### Why was it missed
{Some explanation why this issue might have been missed during normal development/testing cycle}  

### How can we avoid this
{if we don’t want to see this type of issues anymore what we should do to prevent}

### PR DESCRIPTION
**Pull request by 815are**
Fix #856

Problem:
`fs.extendJSON` is used without 4th param `space`:
```
 extendJSON(filepath: string, contents: any, replacer?: WriteJsonReplacer, space?: WriteJsonSpace): void;
```
In result default values is `2`:
image

Solution:
1. New configuration property 'tabInfo' is added, where we an pass tab sizing configuration. Currently it is used for 'manifest.json' file updates;
2. If `tabInfo` is not passed, then logic in [file.ts](https://github.com/SAP/open-ux-tools/blob/fix/856/manifestTabSizing/packages/fe-fpm-writer/src/common/file.ts#L47) tries to calculate current file's tab sizing info. It finds first line with spaces or `\t` and detect if it should use white spaces or `\t` and calculates single tab step.

Additional changes:
1. [list.ts](https://github.com/SAP/open-ux-tools/blob/fix/856/manifestTabSizing/packages/fe-fpm-writer/src/page/list.ts), [object.ts](https://github.com/SAP/open-ux-tools/blob/fix/856/manifestTabSizing/packages/fe-fpm-writer/src/page/object.ts) 
small refactoring to avoid reported sonar duplication issues

### CODE REVIEW

File: .changeset/fifty-schools-hug.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sap-ux/fe-fpm-writer': patch
+---
+
+Fix: writing into manifest.json always provides 2 spaces for tab size. Addition property 'tabInfo' can be passed in config for generatION functions or tab sizing will calculated before saving
```
Comment (stefanschreck) [inline .changeset/fifty-schools-hug.md:5]: Addition ==> Additional
generatION ==> generation
tab sizing will BE calculated

File: packages/fe-fpm-writer/src/common/file.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ * @returns tab size information.
+ */
+export function getJsonSpace(fs: Editor, filePath: string, tabInfo?: TabInfo | undefined): WriteJsonSpace | undefined {
+    let space: WriteJsonSpace | undefined;
+    // Use passed tab info or read from existing content
+    if (!tabInfo) {
+        const content = fs.read(filePath);
+        tabInfo = detectTabSpacing(content);
+    }
+    if (tabInfo !== undefined) {
```
Comment (stefanschreck) [inline packages/fe-fpm-writer/src/common/file.ts:74]: `else` would be sufficient

**Review by stefanschreck — CHANGES_REQUESTED** [review]

File: packages/fe-fpm-writer/src/common/file.ts (hunk 2 above)
Comment (815are) [reply packages/fe-fpm-writer/src/common/file.ts]: @stefanschreck 

I applied some correction in code, but `else` will break logic - result of first if block can affect second if condition. Idea is:
1. 1st condition. if `tabInfo` is not passed - we calculate value for `tabInfo` based on passed file;
2. 2nd condition. It checks if `tabInfo` is defined and calculates value for `space` variable;

We can pass `tabInfo` as custom tab info configuration, but it is optional param. If it is not passed, then first 'if' calculates `tabInfo` by checking target file.

If we add `else`, then calculated value in 1st condition will never be used in second condition, but it should be:
image

I also added some additional tests with new commit to catch such case - https://github.com/SAP/open-ux-tools/pull/857/commits/b234f10d48967c9e5f84bffb7298ffa77c995c00

Please rereview

File: .changeset/fifty-schools-hug.md (hunk 1 above)
Comment (815are) [reply .changeset/fifty-schools-hug.md]: thanks, corrected in edc31bd4a0cb4a5059f18d900c6a747848d09ab4

**Review by zdravko-georgiev — APPROVED** [review]
Code looks good
Very good test coverage
Didn't tested manually

**Review by stefanschreck — APPROVED** [review]

**Pull request merged**